app/services/user.py

```diff
@@ -120,12 +128,22 @@ def update(self) -> User:
                 raise AccessDenied("Wrong password")
             self.user.set_password(self.new_password)
 
+        elif self.email:
+            if User.query.filter_by(email=self.email).first() is not None:
+                raise UserError(detail="Email already in use")
+            self.user.email = self.email
+
         db.session.add(self.user)
         db.session.flush()
 
         return self.user
 
 
+@event.listens_for(User.email, "set")
+def user_notify_email_change(user, *_):
+    UserNotificationService(user).email_changed_email()
```

Comment:
I just realized this should definitely go to the _old_ email.

https://docs.sqlalchemy.org/en/13/orm/events.html

You can get it out of this event notifier in the `old_value` parameter.

Reply:
I added the `old_value` parameter representing the old email and passed it through to `email_changed_email()`